client/core/bond.go

```diff
@@ -259,6 +261,11 @@ func (c *Core) bondStateOfDEX(dc *dexConnection, bondCfg *dexBondCfg) *dexAcctBo
 
 	tierDeficit := int64(state.targetTier) - state.tier
 	state.mustPost = tierDeficit + state.weakStrength - state.pendingStrength
+	// If weak bonds expiring would put the number of live and pending bonds
+	// to zero, replace the weak bonds.
+	if state.mustPost <= 0 && state.weakStrength >= state.liveStrength+state.pendingStrength {
```

Comment:
The liveStrength does not include the bonds in the weakStrength category. Would it work if we just post 1 bond if `state.liveStrength+state.pendingStrength == 0`?

Reply:
Yeah. That may be more correct.

Reply:
The `liveStrength` **does** include weak bonds, so `weakStrength` is always `<= liveStrength`.

Reply:
Ah my mistake.